src/jobs/test-package.yml

```diff
@@ -15,30 +15,51 @@ parameters:
     description: if true, the job will submit code coverage report to codecov.io
     default: false
     type: boolean
+  os:
+    description: operating system you want to test against
+    type: enum
+    enum: ['linux', 'windows']
+    default: 'linux'
 
 executor:
-  name: default
-  tag: << parameters.node_version >>
+  name: << parameters.os >>
 
 steps:
+  - install-node:
+      version: <<parameters.node_version>>
+      os: <<parameters.os>>
   - checkout
-  - restore_cache:
-      keys:
-        - v1-npm-{{checksum ".circleci/config.yml"}}-{{checksum "yarn.lock"}}
-        - v1-npm-{{checksum ".circleci/config.yml"}}
-  - run:
-      name: Install dependencies
-      command: yarn
-  - run:
-      name: Build
-      command: yarn build
-  - save_cache:
-      key: v1-npm-{{checksum ".circleci/config.yml"}}-{{checksum "yarn.lock"}}
-      paths:
-        - ~/node_modules
-        - /usr/local/share/.cache/yarn
-        - /usr/local/share/.config/yarn
-  - verify-installed-plugin
+  - when:
+      condition:
+        equal: ['windows', <<parameters.os>>]
+      steps:
+        - run:
+            name: Install dependencies
```

Comment:
Does windows not support caching? I figured we still need this conditional because the paths are different but it seems like a good thing to have.